server/controllers/finance/fiscal.js

```diff
@@ -297,21 +298,27 @@ function remove(req, res, next) {
  * The balance for a specified fiscal year and period with all accounts
  * the period must be given
  */
-function getBalance(req, res, next) {
-  const { id } = req.params;
-  const period = req.params.period_number || 12;
-  debug(`#getBalance() looking up balance for FY${id} and period ${period}.`);
+async function getBalance(req, res, next) {
+  try {
+    const { id } = req.params;
+    const period = req.params.period_number || 12;
+    debug(`#getBalance() looking up balance for FY${id} and period ${period}.`);
+
+    const result = await sumedBalance(id, period);
+    res.status(200).json(result);
+  } catch (error) {
+    next(error);
+  }
+}
 
-  lookupBalance(id, period)
+function sumedBalance(id, period) {
```

Comment:
Let's call this `summedBalance`.

You can also transform it into an async function to save a few lines of code:
```js
const tree = new Tree(await lookupBalance(id, period));
tree.walk(Tree.common.sumOnProperty('debit'), false);
tree.walk(Tree.common.sumOnProperty('credit'), false);
return tree.toArray();
```